nyc/bills.py

```diff
@@ -1,232 +1,359 @@
+        # Add required URL parameter
+        self.params = {'Token': TOKEN}
 
-    def sessions(self, action_date) :
-        for session in self.SESSION_STARTS :
-            if action_date >= datetime.datetime(session, 1, 1, 
-                                               tzinfo=pytz.timezone(self.TIMEZONE)) :
+
+    def sessions(self, action_date):
+        for session in self.SESSION_STARTS:
+            if action_date >= datetime.datetime(session, 1, 1,
+                                               tzinfo=pytz.timezone(self.TIMEZONE)):
                 return str(session)
 
 
+    def actions(self, matter_id):
+        for action in self.history(matter_id):
+            bill_action = {}
+
+            if not action['MatterHistoryActionName']:
+                continue
+
+            bill_action['action_description'] = action['MatterHistoryActionName']
+            bill_action['action_date'] = action['MatterHistoryActionDate']
+
+            responsible_org = action['MatterHistoryActionBodyName']
+
+            if responsible_org == 'City Council':
+                responsible_org = 'New York City Council'
+
+            elif responsible_org == 'Administration':
+                responsible_org = 'Mayor'
+
+            elif responsible_org == 'Town Hall Meeting':
+                continue
+
+            bill_action['responsible_org'] = responsible_org
+
+            if all(bill_action.values()):
+                action_date = self.toTime(bill_action['action_date']).date()
+                bill_action['action_date'] = action_date
+
+                # Strip trailing newline and carriage return chars
+                classification = ACTION_CLASSIFICATION[bill_action['action_description'].strip()]
+                bill_action['classification'] = classification
+
+                if all(action.get(k, None) for k in ['MatterHistoryEventId',
+                                                     'MatterHistoryRollCallFlag',
+                                                     'MatterHistoryPassedFlag']):
+
+                    bool_result = action['MatterHistoryPassedFlag']
+                    result = 'pass' if bool_result else 'fail'
+
+                    votes = (result, self.votes(action['MatterHistoryId']))
+                else:
+                    votes = (None, [])
+
+                yield bill_action, votes
+
+
+    def _version_rank(self, version):
+        '''
+        - If there is only one version, *, it is the max version.
+        - If there is an asterisk version and one other version, the other
+          version is the max version.
+        - If there is an asterisk version and two other non-numeric versions,
+          the max version is the first other version in alphabetical order.
+        - Otherwise, function as normal.
+        '''
+        version_map = {'*': -1,
+                       'B': 1,
+                       'A': 2}
+
+        return int(version_map.get(version, version))
```

Comment:
let's make this code much more brittle and just return

```python
version_map[version]
```

Reply:
If this method receives a non-integer version we do not expect, it will break. It should function as normal, [returning an int](https://github.com/opencivicdata/python-legistar-scraper/blob/2ba110754b8272b43b95ffd496e24d33e67983a9/legistar/bills.py#L287), with versions that are numbered and hence not in `version_map`.

Reply:
I get it, but that's a lot to ask the reader of the code to know. It's would be better to have all the possibilities that we know about in this code.  For example, what would happen if there was an version value of '4' and a version value of 'A'. That seems possible.

Reply:
Is there a way a la the actions endpoint to quickly get all combinations of versions?

Reply:
Not that I know of.